client/my-sites/comments/comment-list/index.jsx

```diff
@@ -88,21 +89,28 @@ export class CommentList extends Component {
 			this.setState( {
 				isBulkEdit: false,
 				lastUndo: null,
-				page: 1,
 				persistedComments: [],
 				selectedComments: [],
 			} );
 		}
 	}
 
-	changePage = page => {
-		const total = Math.ceil( ( this.props.comments.length + this.state.persistedComments.length ) / COMMENTS_PER_PAGE );
-		this.props.recordChangePage( page, total );
+	changePage = pageNumber => {
+		const {
+			recordChangePage,
+			siteFragment,
+			status,
+		} = this.props;
 
-		this.setState( {
-			page,
-			selectedComments: [],
-		} );
+		recordChangePage( page, this.getTotalPages() );
+
+		this.setState( { selectedComments: [] } );
+
+		if ( window ) {
+			window.scrollTo( 0, 0 );
+		}
+
+		page( addQueryArgs( { page: pageNumber }, `/comments/${ status }/${ siteFragment }` ) );
```

Comment:
I would recommend inverting this a bit. Add the query param to the pager links. 

Make the controller pass down the page prop to this component.

Reply:
Not a blocking comment on this point (since we appear to add search query params in this manner) but there would be two benefits to this:

1. We can reduce this method to fire off an analytics event
2. folks can open the next page in a new tab